crates/server/src/admin/handlers/blob.rs

```diff
@@ -0,0 +1,267 @@
+        }
+    } else {
+        None
+    };
+
+    tracing::info!("Starting streaming raw blob upload");
+
+    // Create streaming reader from the body
+    let reader = body_to_async_read(body);
+
+    // Store the blob using the node client with streaming
+    match state
+        .node_client
+        .add_blob(reader, None, expected_hash.as_ref())
+        .await
+    {
+        Ok((blob_id, size)) => {
+            tracing::info!(
+                "Successfully uploaded streaming blob {} with size {} bytes ({:.1} MB)",
+                blob_id,
+                size,
+                size as f64 / (1024.0 * 1024.0)
+            );
+            ApiResponse {
+                payload: BlobInfo { blob_id, size },
+            }
+            .into_response()
+        }
+        Err(err) => {
+            tracing::error!("Failed to upload streaming blob: {:?}", err);
+            ApiError {
+                status_code: StatusCode::INTERNAL_SERVER_ERROR,
+                message: format!("Failed to store blob: {}", err),
+            }
+            .into_response()
+        }
+    }
+}
+
+/// Download a blob by its ID
+///
+/// Returns the raw binary data of the blob with appropriate content headers.
+pub async fn download_handler(
+    Path(blob_id): Path<String>,
+    Extension(state): Extension<Arc<AdminState>>,
+) -> impl IntoResponse {
+    // Parse blob ID
+    let blob_id: BlobId = match blob_id.parse() {
+        Ok(id) => id,
+        Err(_) => {
+            return ApiError {
+                status_code: StatusCode::BAD_REQUEST,
+                message: "Invalid blob ID format".to_owned(),
+            }
+            .into_response();
+        }
+    };
+
+    // Get blob data
+    match state.node_client.get_blob_bytes(&blob_id).await {
+        Ok(Some(blob_data)) => {
+            // Detect MIME type from content
+            let content_type = detect_mime_type(&blob_data);
+
+            // Generate a more descriptive filename based on content type
+            let filename = match content_type {
+                "image/png" => format!("blob-{}.png", &blob_id.to_string()[..8]),
+                "image/jpeg" => format!("blob-{}.jpg", &blob_id.to_string()[..8]),
+                "image/gif" => format!("blob-{}.gif", &blob_id.to_string()[..8]),
+                "image/webp" => format!("blob-{}.webp", &blob_id.to_string()[..8]),
+                "application/pdf" => format!("blob-{}.pdf", &blob_id.to_string()[..8]),
+                "application/zip" => format!("blob-{}.zip", &blob_id.to_string()[..8]),
+                "text/html" => format!("blob-{}.html", &blob_id.to_string()[..8]),
+                "application/json" => format!("blob-{}.json", &blob_id.to_string()[..8]),
+                "application/xml" => format!("blob-{}.xml", &blob_id.to_string()[..8]),
+                "text/plain" => format!("blob-{}.txt", &blob_id.to_string()[..8]),
+                _ => format!("blob-{}", &blob_id.to_string()[..8]),
+            };
```

Comment:
I'm not convinced we need any of this tbh, store a raw blob, retrieve a raw blob

if the blob is being used in some app, let it be the apps responsibility to keep track of it's metadata